server/route.go

```diff
@@ -491,6 +484,24 @@ func routeSid(sub *subscription) string {
 	return fmt.Sprintf("%s%s:%d:%s", qi, RSID, sub.client.cid, sub.sid)
 }
 
+func parseRouteSid(rsid []byte) (matches [EXPECTED_MATCHES][]byte, ok bool) {
+	if !bytes.HasPrefix(rsid, []byte(QRSID+":")) {
+		return
+	}
+
+	start := len(QRSID + ":")
+	for i := start; i < len(rsid); i++ {
+		switch rsid[i] {
+		case ':':
+			matches[RSID_CID_INDEX], matches[RSID_SID_INDEX] = rsid[start:i], rsid[i+1:]
```

Comment:
If you change to return individual `[]byte`, I would then do `return rsid[start:i], rsid[i+1:], true`.